File: articles/linear-algebra-for-systems-modeling/articles/inverse-matrices-and-structural-recovery/typescript/inverse_recovery.ts
const a = 3;
const b = 1;
const c = 2;
const d = 4;

const y1 = 7;
const y2 = 8;

const det = a * d - b * c;

if (det === 0) {
  console.log("Matrix is singular; recovery is not unique.");
} else {
  const x1 = (d * y1 - b * y2) / det;
  const x2 = (-c * y1 + a * y2) / det;
  console.log(`Recovered state: x1 = ${x1.toFixed(2)}, x2 = ${x2.toFixed(2)}`);
}

File: articles/linear-algebra-for-systems-modeling/articles/inverse-matrices-and-structural-recovery/go/inverse_recovery.go
package main

import "fmt"

func main() {
	a, b, c, d := 3.0, 1.0, 2.0, 4.0
	y1, y2 := 7.0, 8.0
	det := a*d - b*c

	if det == 0 {
		fmt.Println("Matrix is singular; recovery is not unique.")
		return
	}

	x1 := (d*y1 - b*y2) / det
	x2 := (-c*y1 + a*y2) / det

	fmt.Printf("Recovered state: x1 = %.2f, x2 = %.2f\n", x1, x2)
}
